e2e/nomostest/clusters/gke.go

```diff
@@ -73,7 +139,10 @@ func deleteGKECluster(t testing.NTB, name string) error {
 	// Sometimes an operation may be happening at the time the deletion request is
 	// sent, causing delete to error. Retry for a brief period to increase the
 	// chances for the delete operation.
-	took, err := retry.Retry(1*time.Minute, func() error {
+	took, err := retry.Retry(10*time.Minute, func() error {
```

Comment:
if you use wait.BackoffUntil you can pass it the ctx.Done and cancel this with the same context that cancels the operation wait.

Reply:
Looked into this but I don't think it will be that simple, because `BackOffUntil` doesn't accept a function that returns an error. Thus it continues retrying regardless of whether the function succeeds, which is a bit of a different use case.

Reply:
We would need to implement some sort of hybrid approach that uses the `wait.ConditionFunc` of `wait.ExponentialBackoff` and the `stopCh` of `wait.BackoffUntil`

Reply:
As I mentioned `BackOffUntil` doesn't really provide the right interface for this use case. I added a new function which fits this pattern. I played with a few different ideas but didn't want to go too overkill on async/channels so I think this is a good middle ground. PTAL

/unhold